code/aind_ccf_reg/preprocess.py

```diff
@@ -0,0 +1,296 @@
+"""
+Preprocess lightsheet data
+"""
+import logging
+from datetime import datetime
+
+import ants
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy
+from aind_ccf_reg.plots import plot_antsimgs
+
+import scipy.ndimage as ni
+from skimage.filters import threshold_li
+from skimage.measure import label   
+from skimage import io
+import tifffile
+
+from pathlib import Path
+
+from aind_ccf_reg.configs import VMIN, VMAX
+from aind_ccf_reg.configs import PathLike
+
+
+LOG_FMT = "%(asctime)s %(message)s"
+LOG_DATE_FMT = "%Y-%m-%d %H:%M"
+
+logging.basicConfig(format=LOG_FMT, datefmt=LOG_DATE_FMT)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+def perc_normalization(ants_img):
+    """
+    Percentile Normalization 
+    
+    Parameters
+    -------------
+    ants_img: ANTsImage
+        
+    Returns
+    -----------
+    ANTsImage
+    """
+    percentiles = [2, 98]
+    percentile_values = np.percentile(ants_img.view(), percentiles)
+    ants_img = (ants_img - percentile_values[0]) / (percentile_values[1] - percentile_values[0])
+
+    return ants_img
+
+def write_and_plot_image(ants_img, data_path=None, plot_path=None, vmin=VMIN, vmax=VMAX):
+    """ 
+    Write and plot ants image
+
+    Parameters
+    -------------
+    ants_img: ANTsImage
+    data_path: PathLike
```

Comment:
Please see that if _plot_path_ and _data_path_ are both None, the function will do nothing.